animation_nodes/nodes/sound/sound_spectrum.py

```diff
@@ -0,0 +1,145 @@
+import bpy
+import numpy
+from math import expm1
+from bpy.props import *
+from ... base_types import AnimationNode
+from ... data_structures import DoubleList
+
+samplingItems = [
+    ("EXP", "Exponential", "", "", 0),
+    ("CUSTOM", "Custom", "", "", 1),
+    ("SINGLE", "Single", "", "", 2),
+    ("FULL", "Full", "", "", 3)
+]
+reductionItems = [
+    ("MEAN", "Mean", "", "", 0),
+    ("MAX", "Max", "", "", 1)
+]
+reductionFunctions = {
+    "MEAN" : numpy.mean,
+    "MAX" : numpy.max
+}
+
+class SoundSpectrumNode(bpy.types.Node, AnimationNode):
+    bl_idname = "an_SoundSpectrumNode"
+    bl_label = "Sound Spectrum"
+    errorHandlingType = "EXCEPTION"
+
+    reductionFunction: EnumProperty(name = "Reduction Function", default = "MAX",
+        items = reductionItems)
+    smoothingSamples: IntProperty(name = "Smoothing Samples", default = 5, min = 0)
+    beta: FloatProperty(name = "Kaiser Beta", default = 6, min = 0)
+
+    samplingMethod: EnumProperty(name = "Sampling Method", default = "EXP", items = samplingItems,
+        update = AnimationNode.refresh)
+
+    def create(self):
+        self.newInput("Sound", "Sound", "sound")
+        self.newInput("Float", "Frame", "frame")
+        self.newInput("Float", "Attack", "attack", value = 0.005, minValue = 0, maxValue = 1)
+        self.newInput("Float", "Release", "release", value = 0.6, minValue = 0, maxValue = 1)
+        self.newInput("Float", "Amplitude", "amplitude", value = 1)
+
+        if self.samplingMethod == "EXP":
+            self.newInput("Integer", "Count", "count", value = 20, minValue = 1)
+            self.newInput("Float", "Low", "low", value = 0, minValue = 0, maxValue = 1, hide = True)
+            self.newInput("Float", "High", "high", value = 1, minValue = 0, maxValue = 1, hide = True)
+            self.newInput("Float", "K", "k", value = 5, minValue = 0.00001)
+        elif self.samplingMethod == "CUSTOM":
+            self.newInput("Float List", "Pins", "pins")
+        elif self.samplingMethod == "SINGLE":
```

Comment:
I think the `SINGLE` method can be moved completely to the Sound Intensity node (which should also have the high and low input then.

Reply:
I think the user should expect this option to be in the Sound Spectrum node. I associate the *"Sound Intensity"* with the overall sound intensity and not the amplitude of a certain frequency bin. One might think of it as follows: If one wants to sample overlapping frequency bins, one would use a loop with the `SINGLE` option to get the spectral data. Thus the option was used to compute spectral data and not sound intensity, so we ask, what node should be used here? Moreover, the Sound Intensity node is already simple with no enums, so why crowd it with an extra option that can be in another place that is—in my opinion—more appropriate?

So I think the `SINGLE` option should stay in the Sound Spectrum node.

Reply:
I just wonder about the typical workflow. Usually when someone wants to animate a single value using sound, he would use the Sound Intensity node. However often more control is needed, so later in the process he replaces the Sound Intensity node with a Sound Spectrum node.

With this option the Sound Intensity node is redundant and should be removed.
Maybe the best solution is to add an additional search result for the Sound Spectrum node that initializes the Single option directly?

Reply:
This is not the only use for the single option. The single option can be used inside loops to generate frequency bins that are otherwise not possible to generate using the rest of the options, like overlapping frequency bins. So the user might be tempted to think that this is not possible because she is looking in the wrong place, that is, the Sound Spectrum node instead of the Sound Intensity. At least that's where I would look for this option.

That might be a solution, but I am not sure about the relation between the sound intensity and the frequency domain. Will research this and get back to you.

Reply:
Ok, I think the Sound Intensity node can be removed and a search entry can be added.
So what should we do?

Reply:
Remove the Sound Intensity node and add the search entry.

Reply:
Ok, done.

Reply:
I also removed the amplitude parameter and set the default amplitude to 10. You were right, users can easily make use of it in that order of magnitude.